javascripts/controllers/ViewCtrl.js

```diff
@@ -20,4 +20,28 @@ app.controller("ViewCtrl", function($rootScope, $scope, ContactService) {
 		});
 	};
 
+$scope.favoriteContacts = (contact) => {
+	let updatedContact = {};
+
+	if (!contact.favorite) {
+		updatedContact = ContactService.createContactObject(contact);
+	} else {
+		updatedContact = ContactService.createContactObject(contact);
+		updatedContact.favorite = false;
+	}
+	ContactService.updateContact(updatedContact, contact.id).then(() => {
```

Comment:
See the other comment about weird bug happening here.

Reply:
Fixed this bug